packages/sites/src/create-site-model-from-template.ts

```diff
@@ -114,12 +115,9 @@ export function createSiteModelFromTemplate(
         );
         settings.solution.url = getPortalSiteUrl(uniqueSubdomain, portal);
       } else {
-        settings.solution.defaultHostname = `${uniqueSubdomain}-${
-          portal.urlKey
-        }.${stripProtocol(hubRequestOptions.hubApiUrl)}`;
-        settings.solution.url = `https://${uniqueSubdomain}-${
-          portal.urlKey
-        }.${stripProtocol(hubRequestOptions.hubApiUrl)}`;
+        const base = stripProtocol(getHubApiUrl(hubRequestOptions));
```

Comment:
Seems I can't add a comment in the un-changed code - but this fn takes an `IHubRequestOptions` which now has `authorization` as optional... but this fn needs `authorization`...  I guess this is a way to make a non-breaking change, but it also feels like it opens up possible confusion... could we make it `IHubRequestOptions | IHubUserRequestOptions` so we're widening the types, and then at a major we can remove the first one?

Reply:
Good catch.

I thought since those clients were already passing in a `IHubRequestOptions` _with_ `authentication` as a `UserSession` that TS would be able to figure out that's the same as a `IHubUserRequstOptions`, but alas, no. I just tried updating the signature of that fn to take only a `IHubUserRequestOptions` and I got a bunch of compile errors when running the tests about not being able to convert `IHubRequestOptions` to `IHubUserRequestOptions`. 

I then tried your suggestion of `hubRequestOptions: IHubUserRequestOptions | IHubRequestOptions` and tests compiled and passed. So we could do that, but I'd have to figure out all the places that require `authentication` as a `UserSession`. I think that's better than backing out the `authentication` related changes, b/c ultimately we'll want to have `authentication` optional for the majority of requests, so we'll have to do this work at some point.

Reply:
After our chat this AM, I went through Solution.js and learned that it uses these fns from `@esr/hub-sites`:

```
createPageModelFromTemplate,
createPage,
removePage,
updatePage,
_updatePages,

createSiteModelFromTemplate,
createSite,
getSiteById,
removeSite,
convertSiteToTemplate
_shareItemsToSiteGroups,
updateSite
```

I suspect most of those other than `getSiteById()` would need the `IHubUserRequestOptions | IHubRequestOptions` hack in order to prevent TS compile errors in that package if we want to move forward w/ this PR.

I suspect that template.js imports the same fns, but haven't confirmed.